staging/src/k8s.io/component-base/logs/logs.go

```diff
@@ -165,17 +166,41 @@ func (writer KlogWriter) Write(data []byte) (n int, err error) {
 	return len(data), nil
 }
 
+// InitLogOption implements the functional options pattern for InitLogs
+type InitLogOption func(o *initLogOptions)
+
+// FeatureGate can be used to enable or disable features. If this option is not
+// used, the parameter is nil, or a specific feature has not been registered,
+// the default for the feature will be used.
+func FeatureGate(featureGate featuregate.FeatureGate) InitLogOption {
+	return func(o *initLogOptions) {
+		o.featureGate = featureGate
+	}
+}
+
+type initLogOptions struct {
+	featureGate featuregate.FeatureGate
+}
+
 // InitLogs initializes logs the way we want for Kubernetes.
 // It should be called after parsing flags. If called before that,
 // it will use the default log settings.
-func InitLogs() {
+func InitLogs(opts ...InitLogOption) {
```

Comment:
InitLogs again uses functional options to avoid breaking code that doesn't care about passing a FeatureGate.